packages/core/src/context/ContextSystem.ts

```diff
@@ -278,5 +278,22 @@ export class ContextSystem extends System
 
             /* eslint-enable max-len */
         }
+
+        const hasuint32
+            = (gl instanceof WebGL2RenderingContext)
+            || !!(gl as WebGLRenderingContext).getExtension('OES_element_index_uint');
+
+        if (!hasuint32 && settings.HAS_UINT32_INDEX)
+        {
+            /* eslint-disable max-len */
+
+            /* eslint-disable no-console */
+            console.warn('Provided WebGL context does not support 32 index buffer, large graphics may not render correctly');
+            /* eslint-enable no-console */
+
+            /* eslint-enable max-len */
+        }
+
+        settings.HAS_UINT32_INDEX = settings.HAS_UINT32_INDEX && hasuint32;
```

Comment:
I don't like resetting the user's choice here. Settings should be immutable from within pixi. `PIXI.settings.ROUND_PIXELS` is a good example, where it just sets the default behavior, but display objects implement. Maybe just set an internal property on ContextSystem?

Reply:
How we should read settings then, because HAS_UINT32_INDEX reads in GraphicsGeometry ?
There aren't immutable globals setings like @pixi/settings.
This is same problem as MAX_TEXTURES. There are two MAX_TEXTURE: in settings, in renderer, but accesing only from settings.
We need object, somethink like LOCKED_SETTINGS for this, that can't be changed from PIXI.settings after initialisation.

Reply:
We need to store it somewhere and not in renderer. Its close to `getTestContext` and we need refactoring of that part before we put it there.

Reply:
I think we should add an `env` state object either to core (or a new package, or something in settings), that will be a collection of readonly properties for users that can be set from within Pixi. This would contain things like webgl/webgl2, has_uint32_index, max_textures, etc.